crates/iceberg/src/spec/values.rs

```diff
@@ -70,6 +79,571 @@ pub enum PrimitiveLiteral {
+    /// assert_eq!(Literal::time(micro_secs), t.into());
+    ///
+    /// let negative_value = -100;
+    /// assert!(Datum::time(negative_value).is_err());
+    ///
+    /// let too_large_value = 36 * 60 * 60 * 1_000_000; // Too large to fit in 24 hours.
+    /// assert!(Datum::time(too_large_value).is_err());
+    /// ```
+    pub fn time(value: i64) -> Result<Self> {
+        ensure_data_valid!(
+            (0..=MAX_TIME_VALUE).contains(&value),
+            "Invalid value for Time type: {}",
+            value
+        );
+
+        Ok(Self {
+            r#type: PrimitiveType::Time,
+            literal: PrimitiveLiteral::Time(value),
+        })
+    }
+
+    /// Creates time literal from [`chrono::NaiveTime`].
+    fn time_from_naive_time(t: NaiveTime) -> Self {
+        let duration = t - unix_epoch().time();
+        // It's safe to unwrap here since less than 24 hours will never overflow.
+        let micro_secs = duration.num_microseconds().unwrap();
+
+        Self {
+            r#type: PrimitiveType::Time,
+            literal: PrimitiveLiteral::Time(micro_secs),
+        }
+    }
+
+    /// Creates time in microseconds in `%H:%M:%S:.f` format.
+    ///
+    /// See [`NaiveTime::from_str`] for details.
+    ///
+    /// Example:
+    /// ```rust
+    /// use iceberg::spec::{Literal, Datum};
+    /// let t = Datum::time_from_str("01:02:01.888999777").unwrap();
+    ///
+    /// assert_eq!(&format!("{t}"), "01:02:01.888999");
+    /// ```
+    pub fn time_from_str<S: AsRef<str>>(s: S) -> Result<Self> {
+        let t = s.as_ref().parse::<NaiveTime>().map_err(|e| {
+            Error::new(
+                ErrorKind::DataInvalid,
+                format!("Can't parse time from string: {}", s.as_ref()),
+            )
+            .with_source(e)
+        })?;
+
+        Ok(Self::time_from_naive_time(t))
+    }
+
+    /// Creates time literal from hour, minute, second, and microseconds.
+    ///
+    /// See [`NaiveTime::from_hms_micro_opt`].
+    ///
+    /// Example:
+    /// ```rust
+    ///
+    /// use iceberg::spec::{Literal, Datum};
+    /// let t = Datum::time_from_hms_micro(22, 15, 33, 111).unwrap();
+    ///
+    /// assert_eq!(&format!("{t}"), "22:15:33.000111");
+    /// ```
+    pub fn time_from_hms_micro(hour: u32, min: u32, sec: u32, micro: u32) -> Result<Self> {
+        let t = NaiveTime::from_hms_micro_opt(hour, min, sec, micro)
+            .ok_or_else(|| Error::new(
+                ErrorKind::DataInvalid,
+                format!("Can't create time from hour: {hour}, min: {min}, second: {sec}, microsecond: {micro}"),
+            ))?;
+        Ok(Self::time_from_naive_time(t))
+    }
+
+    /// Creates a timestamp from unix epoch in microseconds.
+    ///
+    /// Example:
+    ///
+    /// ```rust
+    ///
+    /// use iceberg::spec::Datum;
+    /// let t = Datum::timestamp(1000);
+    ///
+    /// assert_eq!(&format!("{t}"), "1970-01-01 00:00:00.001");
+    /// ```
+    pub fn timestamp(value: i64) -> Self {
```

Comment:
Same here:
```suggestion
    pub fn timestamp_micros(value: i64) -> Self {
```

Nanoseconds will be added soon 👍